### PR TITLE
feat(TCOMP-230): add flow varibles support

### PR DESCRIPTION
**Pull request by gonchar-ivan**
Related to https://jira.talendforge.org/browse/TCOMP-230

### CODE REVIEW

File: main/plugins/org.talend.designer.codegen/jet_stub/generic/component_main.javajet [hunk 1]
```diff
@@ -142,8 +142,10 @@ if(hasInput){
             <%
             if (mainIrToRow != null) {
                 %>
-            	if(writer_<%=cid%> instanceof org.talend.components.api.component.runtime.WriterWithFeedback<?, ?, ?>){
-                    java.util.Iterator<?> outgoingMainRecords_<%=cid%> = ((org.talend.components.api.component.runtime.WriterWithFeedback<?, ?, ?>)writer_<%=cid%>).getSuccessfulWrites().iterator();
+            	if (writer_<%=cid%> instanceof org.talend.components.api.component.runtime.WriterWithFeedback<?, ?, ?>) {
+					org.talend.components.api.component.runtime.WriterWithFeedback<?, ?, ?> writerWithFeedbackMain_<%=cid%> = (org.talend.components.api.component.runtime.WriterWithFeedback<?, ?, ?>) writer_<%=cid%>;
+					writerWithFeedbackMain_<%=cid%> = new org.talend.codegen.flowvariables.runtime.FlowVariablesWriter<Object>(writerWithFeedbackMain_<%=cid%>, container_<%=cid%>);
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.codegen/jet_stub/generic/component_main.javajet]: some compiler issue : The constructor FlowVariablesWriter<Object>(WriterWithFeedback<capture#12-of ?,capture#13-of ?,capture#14-of ?>, RuntimeContainer) is undefined, please check it.

Comment (gonchar-ivan) [inline main/plugins/org.talend.designer.codegen/jet_stub/generic/component_main.javajet]: Hm... Thanks. I'll check it.

**Comment by gonchar-ivan** [comment]
Sorry. Seems, I haven't check it with Output component. That's why I missed compilation error.
After fixing compilation error I found one more bug in codegen-utils library itselft. I made a fix for it.
@wwang-talend could you please take a look on that fix:
https://github.com/Talend/tdi-studio-se/pull/983

983 PR should be merged and integrated before this PR

**Comment by wwang-talend** [comment]
@gonchar-ivan will check #983

**Pull request merged**